packages/core/src/agent/index.ts

```diff
@@ -120,7 +121,11 @@ export class Agent<
     if (config.voice) {
       this.voice = config.voice;
       this.voice?.addTools(this.tools);
-      this.voice?.addInstructions(config.instructions);
+      if (typeof config.instructions === "string") {
```

Comment:
I'm not sure what the solution should do here. Seems like an external provider is safer when we force this to be a string for voice agents.